src/ocean/net/ssl/openssl/OpenSsl.d

```diff
@@ -0,0 +1,833 @@
+
+        This function should never be called directly. The documentation
+        includes three pages of warnings, caveats, and bugs, noting that unless
+        great care is taken with the parameters, this function causes segfaults
+        and/or internal memory corruption.
+
+        Use decodeRSAPrivateKey() instead.
+
+    ***************************************************************************/
+
+    private RSA * d2i_RSAPrivateKey (RSA ** ptr_to_ptr_which_must_be_null,
+        Const!(void)** ptr_to_ptr_which_gets_corrupted, c_long len);
+
+}
+
+
+/*******************************************************************************
+
+    SSL error codes
+
+*******************************************************************************/
+
+public enum
+{
+    SSL_ERROR_NONE = 0,
+    SSL_ERROR_SSL  = 1,
+    SSL_ERROR_WANT_READ = 2,
+    SSL_ERROR_WANT_WRITE = 3,
+    SSL_ERROR_WANT_X509_LOOKUP = 4,
+    SSL_ERROR_SYSCALL = 5,
+    SSL_ERROR_ZERO_RETURN = 6,
+    SSL_ERROR_WANT_CONNECT = 7,
+    SSL_ERROR_WANT_ACCEPT = 8
+}
+
+
+/*******************************************************************************
+
+    Enum used by SSL_CTX_set_verify
+
+*******************************************************************************/
+
+enum
+{
+    SSL_VERIFY_NONE = 0,
+    SSL_VERIFY_PEER = 1
+}
+
+
+/*******************************************************************************
+
+    Adds the options to the SSL context. Options already set before are not
+    cleared.
+
+    Params:
+        ctx = the SSL context
+        op = the option flags to set.
+
+    Returns:
+        The new options bitmask, after adding options.
+
+*******************************************************************************/
+
+public c_long SSL_CTX_set_options (SSL_CTX* ctx, c_long op)
+{
+    const SSL_CTRL_OPTIONS = 32;
+
+    return SSL_CTX_ctrl(ctx, SSL_CTRL_OPTIONS, op, null);
+}
+
+
+/*******************************************************************************
+
+    Options used by SSL_CTX_set_options
+
+*******************************************************************************/
+
+public enum : ulong
+{
+    SSL_OP_ALL  = 0x80000BFFL,  /// Various workarounds for broken implementations
+    SSL_OP_NO_SSLv2 = 0x01000000,  /// Do not use SSL v2
+    SSL_OP_NO_SSLv3 = 0x02000000,   /// Do not use SSL v3.
+    SSL_OP_NO_COMPRESSION = 0x00020000  /// Do not use compression
+}
+
+
+/*******************************************************************************
+
+    Decodes a PKCS#1 RSAPrivateKey structure, creating an RSA object
+
+    Params:
+        key = the bytes to decode
+
+    Returns:
+        A decoded structure, or NULL if an error occurs.
+
+*******************************************************************************/
+
+public RSA * decodeRSAPrivateKey (Const!(void)[] key)
+{
+    // If this isn't null, memory corruption will happen
+
+    RSA * rsa = null;
+
+    // This variable gets corrupted by the call
```

Comment:
What do you mean corrupted ? Does it expects an lvalue because it zeroes it ?

Reply:
I mean it is set to a nonsense value by the function call. It's obscene.